### DISCUSSION TITLE
LibCarla\source\carla\road\MapBuilder.cpp, Assertion Errors

### DISCUSSION THREAD

**Discussion by pavow** [carla-simulator/carla#5802] [Q&A]
Hi,

Background is that I completed:
  1) A MS Windows 10 source build of Carla 0.9.13 with Visual Studio 2019.
  2) Imported a vendor provided ".xodr" file (OpenDrive Version 1.5) into RoadRunner, then used the export feature of RoadRunner to generate a ".fbx" and".xodr" file (version 1.5) among others.  No modifications were made within RoadRunner.

I sought to create a custom map for Carla by moving the files created in #2 to C:\Carla\Import and executing:

`make import`

This failed with popup error message:
`...Plugins\Carla\Binaries\Win64\UE4Editor.Carla.dll" File C:\carla\LibCarla\source\carla\road\MapBuilder.cpp Line 746   Expression: next_lane != nullptr ....`

I tried again, but this time using the vendor provided ".xodr" file instead of the one exported from RoadRunner.

`make import ARGS="--package=t03"`

This failed with popup error message:
`...Plugins\Carla\Binaries\Win64\UE4Editor.Carla.dll" File C:\carla\LibCarla\source\carla\road\MapBuilder.cpp Line 625 Expression: section != nullptr ....`

I am not familiar with the Carla source code, but am familiar with C and python.  Are the above messages indicating that these ".xodr" files are not complaint with Carla requirements?  If so, are the Carla requirements for xodr files documented anywhere?

Thanks
Pete VW

**Comment by KuangSheng34** [comment]
Hey guys, I met the same problem as you, have you solved this problem?

**Comment by pavow** [comment]
This problem was never solved.

**Reply by KuangSheng34** [reply]
Hi bro, I think I might have found the issue. CARLA seems to use `uint32_t` internally for `RoadId`, so the max value is around **4294967295**. Maybe the road IDs in your .xodr file are too large.
After I fixed the road IDs / mapped them to smaller values, the import worked successfully.

**Reply by KuangSheng34** [reply]
Btw, it may not only be `RoadId`. Other IDs like `JunctionId` might have the same issue, so it’s probably worth checking all large ID values in the `.xodr` file.

**Reply by JArmandoAnaya** [reply]
Hello @KuangSheng34, interesting finding. If you have a chance, it would be great if you could open an issue adding your findings, and I can take a look later. Or you can send a PR if you feel you already have the solution. 👋

**Reply by KuangSheng34** [reply]
Hello @JArmandoAnaya, the issue seems a bit more complicated than I expected. I’m currently investigating under what conditions large `RoadId` values trigger this error. Once I figure it out, I’ll open an issue with the details.

**Comment by pavow** [comment]
Thanks.  I will look into that if we ever use this simulator in the future.
